react/features/base/media/middleware.js

```diff
@@ -68,6 +68,32 @@ function _setRoom({ dispatch, getState }, next, action) {
     const state = getState();
     const { room } = action;
     const roomIsValid = isRoomValid(room);
+    const { error } = state['features/base/conference'];
+    const { audioOnly: previousAudioOnly } = state['features/base/conference'];
+    const { audio, video } = state['features/base/media'];
+
+    let audioMuted, facingMode, videoMuted;
+
+    // startAudioOnly
+    //
+    // FIXME Technically, the audio-only feature is owned by base/conference,
+    // not base/media so the following should be in base/conference.
+    // Practically, I presume it was easier to write the source code here
+    // because it looks like startWithAudioMuted and startWithVideoMuted.
+    //
+    // XXX After the introduction of the "Video <-> Voice" toggle on the
+    // WelcomePage, startAudioOnly is utilized even outside of
+    // conferences/meetings.
+    let audioOnly;
+
+    if (room && error) {
+        // This is a conference rejoin, so we want to reuse the values that are
+        // in 'features/base/media'
+        audio && (audioMuted = audio.muted);
```

Comment:
If it makes a difference, it can be on web although web likely won't go through this code path. Maybe it's worthwhile to leave a comment about how web doesn't exercise this flow?